[Zhen Xuan] iP

zhenxuantan opened the pull request:
## Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke helps you to get a hang of your life. It is
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is 
1. download [here](https://github.com/zhenxuantan/ip/releases).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😜 

And it is **FREE** with **NO INSTALLATION NEEDED**!

Features:
- [x] Add tasks
- [x] List tasks and at a specific date
- [x] Mark tasks as done
- [x] Delete tasks
- [x] Find tasks
- [ ] GUI (Prettify 😃)

 Description | Command | Example
----- | ----- | ----- |
Add a todo task | `todo`**_`description`_** | `todo CS2100 Tutorial`
Add a deadline task | `deadline`**_`description /by date (YYYY-MM-DD HHmm`_** | `deadline ES2660 ReFlEcTiOn /by 2021-08-29 2359`
Add a event task | `event`**_`description /at date (YYYY-MM-DD HHmm`_** | `event CS2100 Finals /at 2021-11-20 0900`
Exit the application | `bye` | `bye`
List down all of the tasks | `list` | `list`
List down all of the deadlines / events on that date | `list`**_`YYYY-MM-DD`_** | `list 2021-11-20`
Delete task (e.g. 2) | `delete`**_`##`_** | `delete 2`
Mark task (e.g. 1) as done | `done`**_`##`_** | `done 1`
Find task with the given words | `find`**_`word(s)`_** | `find CS2100`

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method in the `Duke` class:

```java
public class Duke {
    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

zhenxuantan committed:
Complete Level-1

zhenxuantan committed:
Complete Level-2

zhenxuantan committed:
Complete Level-3

zhenxuantan committed:
Complete Level-4, A-Classes, A-Inheritance

zhenxuantan committed:
Complete A-TextUiTesting, Restructure for A-Classes and A-Inheritance

zhenxuantan committed:
Complete Level-5 and A-Exceptions

zhenxuantan committed:
Complete Level-6 and A-Collections

zhenxuantan committed:
Complete A-Enums

zhenxuantan committed:
Complete Level-7

zhenxuantan committed:
Complete Level-8

zhenxuantan committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Complete Level-8

# Conflicts:
#	.gitignore
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/TaskList.java

zhenxuantan committed:
Complete A-MoreOOP

zhenxuantan committed:
Complete A-Packages

zhenxuantan committed:
Add PrintWriter, Rely more on Ui

zhenxuantan committed:
Complete J-Unit

zhenxuantan committed:
Complete A-JUnit

zhenxuantan committed:
Fix NullPointerException

zhenxuantan committed:
Complete A-JavaDoc

zhenxuantan committed:
Complete A-CodingStandard

zhenxuantan committed:
Complete Level-9

Make tasklist more user-friendly by giving search feature.

Add the particular feature.

Adding the feature gives more flexibility to Duke.

zhenxuantan committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Complete A-CodingStandard

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/command/ByeCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeadlineCommand.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/DoneCommand.java
#	src/main/java/duke/command/EventCommand.java
#	src/main/java/duke/command/InvalidCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/ToDoCommand.java
#	src/test/java/command/DeadlineCommandTest.java
#	src/test/java/command/DeleteCommandTest.java
#	src/test/java/command/DoneCommandTest.java

zhenxuantan committed:
Merge tag 'Level-9'

* tag 'Level-9':
  Complete Level-9

# Conflicts:
#	src/main/java/duke/Parser.java

zhenxuantan committed:
Edit JavaDocs

zhenxuantan committed:
Complete A-Gradle

zhenxuantan committed:
Revert "Complete A-Gradle"

This reverts commit 0951d24ce49f8f40ffc9aae997f0860b26eddd3b.

zhenxuantan committed:
Revert "Revert "Complete A-Gradle""

This reverts commit 3cb1778bf2a0ad95383aa3037b1d5fce476c98d7.

zhenxuantan committed:
Merge branch 'add-gradle-support' of https://github.com/zhenxuantan/ip into add-gradle-support

# Conflicts:
#	build.gradle
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

zhenxuantan committed:
Add Gradle support

zhenxuantan committed:
Complete A-CheckStyle

zhenxuantan committed:
Add GUI

zhenxuantan committed:
Add xfml file

trxe commented on src/main/java/duke/tasklist/TaskListDuke.java:
```diff
@@ -0,0 +1,140 @@
+    private static final String ADD = "\t Got it. I've added this duke.task:";
+    private static final String LIST_INTRO = "\t Here are the tasks in your list:";
+    private static final String DONE = "\t Nice! I've marked this duke.task as done:\n\t   ";
+    private static final String DELETE = "\t Noted. I've removed this duke.task:\n\t   ";
```
Did you mean to have `"task"` instead of `"duke.task"` in the printed strings? Same goes for the other `"duke.task"` in this file.

zhenxuantan committed:
Complete Level-10

zhenxuantan committed:
Stage files

zhenxuantan committed:
Improve style

trxe commented on src/main/java/duke/storage/StorageDuke.java:
```diff
@@ -0,0 +1,115 @@
+        }
+
+        //Check if the file is present, else, create it.
+        ArrayList<Task> tl = new ArrayList<>();
```
You could consider renaming `tl` to `tasklist` since the scope of `tl` is not limited to a nested loop.

trxe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+public class Duke {
+    private StorageDuke storage;
+    private TaskListDuke tasks;
+    private final String filepath = "data/tasks.txt";
```
Since `filepath` is technically a constant, it should be in uppercase (`FILEPATH`). I think the coding standard refers to all `final` variables as constants [(here)](https://se-education.org/guides/conventions/java/intermediate.html#naming).

trxe commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,42 @@
+ */
+public abstract class Command {
+
+    protected final String description;
```
Similarly `DESCRIPTION` for this `final` variable (same for the rest of the `final` variables here).

trxe reviewed (commented):
Hi, your code style-wise is mostly ok other than the formatting for your `final` but non-`static` variables, I think they should be in caps too according to the coding standard in this module. Otherwise its mostly legit, nitpicked here and there only. cheers

chongjunwei commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,67 @@
+ * @version CS2103T AY21/22 S1
+ */
+public abstract class Task {
+    protected final String description;
```
The final variable is a constant, and should be in uppercase.

chongjunwei commented on src/main/java/duke/tasklist/TaskListDuke.java:
```diff
@@ -0,0 +1,139 @@
+     * Adds the task into the list.
+     *
+     * @param task the task to be added into the list
+     */
```
I think you should add a parameter description for the returned String (eg @return Current size of the list).

chongjunwei reviewed (commented):
Looks good, just some minor changes needed to adhere to the given coding standard.

zhenxuantan committed:
Improve code quality

Fellow developers have cited coding style violations and that
it needs to conform with the standard to ensure that codes look like
it is bye a single developer.

Final variable names are capitalised and some of the abbreviations
in the variable names are replaced by their full form.

This is so that the code conforms to the coding standards.

A new jar file is released too with all of the JavaFX add-ins.

zhenxuantan committed:
Add Java assertions

The code mostly depends on a few assertions and if the assertions
failed, Duke will not work. Java Assertions prevent this by reducing
the amount of bugs through alerts while compiling.

Assertions are added at different junctions of the code, mostly the
TaskList as that is where most operations are occurring.

Let's,
* add assertions wherever possible in the future
* not add assertions just to replace exceptions
* not add assertions just to replace JUnit assertions

zhenxuantan committed:
Improve Code Quality

zhenxuantan committed:
Add Stream support

zhenxuantan committed:
Merge pull request #1 from zhenxuantan/branch-A-Assertions

Add Java assertions

zhenxuantan committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add Java assertions

zhenxuantan committed:
Merge pull request #2 from zhenxuantan/branch-A-CodeQuality

Improve code quality

zhenxuantan committed:
Merge branch 'master' into branch-A-Streams

zhenxuantan committed:
Merge pull request #3 from zhenxuantan/branch-A-Streams

Add Stream support

zhenxuantan committed:
Use GitHub Actions to set up Continuous Integration (CI)

zhenxuantan committed:
Add Priority to tasks

zhenxuantan committed:
Improve Ui, fix bugs, add user guide

zhenxuantan committed:
Edit user guide

zhenxuantan committed:
Set theme jekyll-theme-tactile

zhenxuantan committed:
Set theme jekyll-theme-slate

zhenxuantan committed:
Edit user guide

zhenxuantan committed:
Merge commit '064c2a1441f658b8837bd457df42a175bcf8c845'

* commit '064c2a1441f658b8837bd457df42a175bcf8c845':
  Set theme jekyll-theme-slate
  Set theme jekyll-theme-tactile

zhenxuantan committed:
Edit user guide

zhenxuantan committed:
Release Duke

zhenxuantan committed:
Improve code quality
